src/Mvc/Mvc.Core/src/JsonOptions.cs

```diff
@@ -37,5 +39,15 @@ public class JsonOptions
         // from deserialization errors that might occur from deeply nested objects.
         // This value is the same for model binding and Json.Net's serialization.
         MaxDepth = MvcOptions.DefaultMaxModelBindingRecursionDepth,
+
+        // The JsonSerializerOptions.GetTypeInfo method is called directly and needs a defined resolver
+        // setting the default resolver (reflection-based) but the user can overwrite it directly or calling
+        // .AddContext<TContext>()
+        TypeInfoResolver = CreateDefaultTypeResolver()
     };
+
+    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026",
+        Justification = "The default type resolver will be set as DefaultJsonTypeInfoResolver (reflection-based) that has the same runtime behavior when TypeResolver is null.")]
+    private static IJsonTypeInfoResolver? CreateDefaultTypeResolver()
```

Comment:
As part of #45527 this method will be updated to use a Feature switch and avoid the DefaultJsonTypeInfoResolver in AOT.

Reply:
Per feedback in https://github.com/dotnet/aspnetcore/pull/45405/files/84b7eb58d36f7ce0a0e2441f09f093955d1024dd#r1046412455, the public property in line 36 should be marked `RequiresUnreferencedCode`.

Reply:
https://github.com/dotnet/aspnetcore/pull/45405#discussion_r1050010407